test/common/common/utility_fuzz_test.cc

```diff
@@ -53,6 +53,13 @@ DEFINE_FUZZER(const uint8_t* buf, size_t len) {
       StringUtil::cropRight(string_buffer.substr(0, split_point),
                             string_buffer.substr(split_point));
     }
+    {
+      const std::string string_buffer(reinterpret_cast<const char*>(buf), len);
+      // sample random bit to use as the whitespace flag
+      bool trimWhitespace = split_point & 1;
+      StringUtil::findToken(string_buffer, string_buffer.substr(0, split_point),
+                            string_buffer.substr(split_point), trimWhitespace);
```

Comment:
I might just be saying this for extra practice, but it looks like you're always finding the token SECOND_HALF in the string
$DELIMITER$SECOND_HALF

(possible $SECOND_HALF contains more delimiters, and will get split)

I think this leads to some variable coupling, maybe it's good to peel off part of the string as source, part as delimiter, and part as token?

If you want to use an advanced fuzzing library for this, we have FuzzedDataProvider :) 
https://github.com/google/fuzzing/blob/master/docs/split-inputs.md

Reply:
I guess I could also do something similar to what is already being done and use `reinterpret_cast<const uint8_t*>(buf)[1] % len` as a second split point (after checking if the length was greater than 1).  Would this be ok?

Reply:
sounds good! thanks